### PR TITLE
[bitnami/thanos] Allow setting the index-cache configuration in a manually created config secret

### PR DESCRIPTION
**Pull request by jotamartos**
### Description of the change

This PR allows configuring the index-cache.yml and bucket-cache.yml files in a existing secret (previously it only possible to configure the objstore.yml file). If you do not use the existingConfigSecret, a secret is generated with the content of the objstoreConfig, indexCacheConfig and bucketCacheConfig so we follow the same approach in both scenarios.

### Benefits

You can now create a secret with the content of the 3 files

### Possible drawbacks

None

### Applicable issues

- fixes https://github.com/bitnami/charts/issues/16046

### Checklist

- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [X] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami-labs/readme-generator-for-helm)
- [X] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [X] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

File: bitnami/thanos/Chart.yaml [hunk 1]
```diff
@@ -28,4 +28,4 @@ name: thanos
 sources:
   - https://github.com/bitnami/containers/tree/main/bitnami/thanos
   - https://thanos.io
-version: 12.5.1
+version: 12.6.0
```
Comment (jotamartos) [inline bitnami/thanos/Chart.yaml:31]: As I am editing an existing variable, and not adding a new one, is this a minor release or a major one?

File: bitnami/thanos/README.md [hunk 2]
```diff
@@ -1250,7 +1250,7 @@ This helm chart supports using custom Objstore configuration.
 
 You can specify the Objstore configuration using the `objstoreConfig` parameter.
 
-In addition, you can also set an external Secret with the configuration file. This is done by setting the `existingObjstoreSecret` parameter. Note that this will override the previous option. If needed you can also provide a custom Secret Key with `existingObjstoreSecretItems`, please be aware that the Path of your Secret should be `objstore.yml`.
+In addition, you can also set an external Secret with the configuration file. This is done by setting the `existingConfigSecret` parameter. Note that this will override the previous option. If needed you can also provide a custom Secret Key with `existingConfigSecretItems`, please be aware that the Path of your Secret should be `objstore.yml`.
```
Comment (fmulero) [inline bitnami/thanos/README.md:1253]: What about `bucket-cache.yml`, `index-cache.yml`? Cache config files should have those keys

**Review by fmulero — COMMENTED** [review]
Great @jotamartos!

I left some concerns/doubts and I have a general a one, Are those config files large? Individual secrets are limited to 1MiB in size.

File: bitnami/thanos/templates/receive/statefulset.yaml [hunk 3]
```diff
@@ -123,7 +123,7 @@ spec:
             - --grpc-address=0.0.0.0:10901
             - --http-address=0.0.0.0:10902
             - --remote-write.address=0.0.0.0:19291
-            {{- if or .Values.objstoreConfig .Values.existingObjstoreSecret }}
+            {{- if or .Values.objstoreConfig .Values.existingConfigSecret }}
```
[the remark this answers is not in the document]
Comment (jotamartos) [reply bitnami/thanos/templates/receive/statefulset.yaml:126]: It's a must to have. That's the minimum but I'll check if we are currently verifying that in a different place.

File: bitnami/thanos/templates/compactor/_pod-template.tpl [hunk 4]
```diff
@@ -204,8 +204,8 @@ spec:
     - name: objstore-config
```
[the remark this answers is not in the document]
Comment (jotamartos) [reply bitnami/thanos/templates/compactor/_pod-template.tpl:204]: Agree, will take a look at this.

**Pull request closed without merging**